include/depthai/utility/Path.hpp

```diff
@@ -0,0 +1,199 @@
+
+   public:
+    /**
+     * @brief Construct Path object from source.
+     *
+     * @param source utf-8 character sequence; converts to Windows utf-16
+     */
+    Path(const std::string& source) : _nativePath(convert_utf8_to_wide(source)) {}
+
+    /**
+     * @brief Construct Path object from source.
+     *
+     * @param source pointer to null-terminated utf-8 character sequence; converts to Windows utf-16
+     */
+    Path(const char* source) : _nativePath(convert_utf8_to_wide(std::string(source))) {}
+
+    /**
+     * @brief Get path in narrow multibyte representation in the current C locale.
+     *
+     *        Will throw exception if there is no valid conversion.
+     *        Could be used to get the narrow ascii representation.
+     *
+     * @return std::string in narrow multibyte representation
+     */
+    std::string string() const;
+
+    #if defined(__cpp_lib_char8_t)
+    /**
+     * @brief Get path in utf-8.
+     *
+     *        Will throw exception if there is no valid conversion.
+     *
+     * @return std::u8string in utf-8
+     */
+    std::u8string u8string() const {
+        return std::filesystem::path(_nativePath).u8string();
+    }
+    #else
+    /**
+     * @brief Get path in utf-8.
+     *
+     *        Will throw exception if there is no valid conversion.
+     *
+     * @return std::string in utf-8
+     */
+    std::string u8string() const;
+    #endif
+#else
+    /**
+     * @brief Get path in native-encoding string; no conversion.
+     *
+     * @return std::string
+     */
+    std::string string() const {
+        return _nativePath;
+    }
+
+    #if defined(__cpp_lib_char8_t)
+    /**
+     * @brief Get path in utf-8.
+     *
+     * @return std::u8string in utf-8
+     */
+    std::u8string u8string() const {
+        return std::filesystem::path(_nativePath).u8string();
+    }
+    #else
+    /**
+     * @brief Get path in utf-8.
+     *
+     * @return std::string in utf-8
+     */
+    std::string u8string() const {
+        return _nativePath;
+    }
+    #endif
+#endif
+
+    /**
+     * @brief Implicitly convert to native-encoding string, suitable for use with OS APIs
+     *
+     * @return std::string of utf-8 on most OSs, std::wstring of utf-16 on Windows
+     */
+    operator string_type() const noexcept {
+        return _nativePath;
+    }
+
+    /**
+     * @brief Returns native-encoding string by const reference, suitable for use with OS APIs
+     *
+     * @return const std::string& of utf-8 on most OSs, const std::wstring& of utf-16 on Windows
+     */
+    const string_type& native() const noexcept {
+        return _nativePath;
+    }
+
+    /**
+     * @brief Observes if path is empty (contains no string/folders/filename)
+     *
+     * @return bool true if the path is empty, false otherwise
+     */
+    DEPTHAI_NODISCARD bool empty() const noexcept {
+        return _nativePath.empty();
+    }
+
+   private:
+    string_type _nativePath;
```

Comment:
```suggestion
    string_type nativePath;
```

Nitpick - unless it follows some other source

Reply:
I did it for usability and to distinguish it from `std::filesystem::native()`.

`dai::path::_nativePath` is a private member which no dev can access. Some coding tools (like vscode) still show private members in their "intellisense". As I was writing/testing, I found myself often trying to use `nativePath` because it was listed and similar to `std::fs::native`. I considered changing to a different noun but I couldn't come up with another word that worked as well. So I used an underscore which completely separates it in Intellisense and visually suggests this is something internal/temp.

I don't recommend `nativePath`. Do you have another noun you prefer? It can NOT be: native, path, directory, nativePath, nativeDirectory, etc. I think it could be "folder" but...yuck 😝

Reply:
pathway, trail?